File: JXPagerViewExample-OC/JXPagerViewExample-OC/Example/Common/TestTableViewCell.m
//
//  TestTableViewCell.m
//  JXPagerViewExample-OC
//
//  Created by jiaxin on 2018/9/12.
//  Copyright © 2018年 jiaxin. All rights reserved.
//

#import "TestTableViewCell.h"

@interface TestTableViewCell ()
@property (nonatomic, strong) UIButton *bgButton;
@end

@implementation TestTableViewCell

- (void)dealloc
{
    self.bgButtonClicked = nil;
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder {
    self = [super initWithCoder:aDecoder];
    if (self) {
        [self initializeViews];
    }
    return self;
}

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        [self initializeViews];
    }
    return self;
}

- (void)initializeViews {
    self.bgButton = [UIButton buttonWithType:UIButtonTypeSystem];
    [self.bgButton addTarget:self action:@selector(bgButtonClicked:) forControlEvents:UIControlEventTouchUpInside];
    [self.contentView addSubview:self.bgButton];
}

- (void)layoutSubviews {
    [super layoutSubviews];

    self.bgButton.frame = self.contentView.bounds;
}

- (void)bgButtonClicked:(UIButton *)btn {
    self.bgButtonClicked();
}

@end

File: Examples/JXPagingViewExample/JXPagingViewExample/Example/Common/TestTableViewCell.swift
//
//  TestTableViewCell.swift
//  JXPagingView
//
//  Created by jiaxin on 2018/9/10.
//  Copyright © 2018年 jiaxin. All rights reserved.
//

import UIKit

class TestTableViewCell: UITableViewCell {
    var bgButtonClicked: (()->())?
    private var bgButton: UIButton?

    deinit {
        bgButtonClicked = nil
    }

    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)

        initializeViews()
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)

        initializeViews()
    }

    func initializeViews() {
        bgButton = UIButton(type: .custom)
        contentView.addSubview(bgButton!)
        bgButton?.addTarget(self, action: #selector(bgButtonClicked(btn:)), for: .touchUpInside)
    }

    override func layoutSubviews() {
        super.layoutSubviews()

        bgButton?.frame = contentView.bounds
    }

    @objc func bgButtonClicked(btn: UIButton) {
        bgButtonClicked?()
    }

    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)

        // Configure the view for the selected state
    }

}
